src/nvim/change.c

```diff
@@ -0,0 +1,1805 @@
+  int ind_done;
+  int round;
+
+  // Round 1: compute the number of characters needed for the indent
+  // Round 2: copy the characters.
+  for (round = 1; round <= 2; round++) {
+    todo = size;
+    ind_len = 0;
+    ind_done = 0;
+    s = src;
+
+    // Count/copy the usable portion of the source line.
+    while (todo > 0 && ascii_iswhite(*s)) {
+      if (*s == TAB) {
+        tab_pad = (int)curbuf->b_p_ts
+                  - (ind_done % (int)curbuf->b_p_ts);
+
+        // Stop if this tab will overshoot the target.
+        if (todo < tab_pad) {
+          break;
+        }
+        todo -= tab_pad;
+        ind_done += tab_pad;
+      } else {
+        todo--;
+        ind_done++;
+      }
+      ind_len++;
+
+      if (p != NULL) {
+        *p++ = *s;
+      }
+      s++;
+    }
+
+    // Fill to next tabstop with a tab, if possible.
+    tab_pad = (int)curbuf->b_p_ts - (ind_done % (int)curbuf->b_p_ts);
+
+    if ((todo >= tab_pad) && !curbuf->b_p_et) {
+      todo -= tab_pad;
+      ind_len++;
+
+      if (p != NULL) {
+        *p++ = TAB;
+      }
+    }
+
+    // Add tabs required for indent.
+    while (todo >= (int)curbuf->b_p_ts && !curbuf->b_p_et) {
+      todo -= (int)curbuf->b_p_ts;
+      ind_len++;
+
+      if (p != NULL) {
+        *p++ = TAB;
+      }
+    }
+
+    // Count/add spaces required for indent.
+    while (todo > 0) {
+      todo--;
+      ind_len++;
+
+      if (p != NULL) {
+        *p++ = ' ';
+      }
+    }
+
+    if (p == NULL) {
+      // Allocate memory for the result: the copied indent, new indent
+      // and the rest of the line.
+      line_len = (int)STRLEN(get_cursor_line_ptr()) + 1;
+      assert(ind_len + line_len >= 0);
+      size_t line_size;
+      STRICT_ADD(ind_len, line_len, &line_size, size_t);
+      line = xmalloc(line_size);
+      p = line;
+    }
+  }
+
+  // Append the original line
+  memmove(p, get_cursor_line_ptr(), (size_t)line_len);
+
+  // Replace the line
+  ml_replace(curwin->w_cursor.lnum, line, false);
+
+  // Put the cursor after the indent.
+  curwin->w_cursor.col = ind_len;
+  return true;
+}
+
+/// open_line: Add a new line below or above the current line.
+///
+/// For VREPLACE mode, we only add a new line when we get to the end of the
+/// file, otherwise we just start replacing the next line.
+///
+/// Caller must take care of undo.  Since VREPLACE may affect any number of
+/// lines however, it may call u_save_cursor() again when starting to change a
+/// new line.
+/// "flags": OPENLINE_DELSPACES delete spaces after cursor
+///          OPENLINE_DO_COM    format comments
+///          OPENLINE_KEEPTRAIL keep trailing spaces
+///          OPENLINE_MARKFIX   adjust mark positions after the line break
+///          OPENLINE_COM_LIST  format comments with list or 2nd line indent
+///
+/// "second_line_indent": indent for after ^^D in Insert mode or if flag
+///                       OPENLINE_COM_LIST
+///
+/// @return true on success, false on failure
```

Comment:
This will cause "readability/fn_size" with linting still.
Can it be explicitly ignored, e.g. with an inline comment?

> src/nvim/change.c:1741:  Small and focused functions are preferred: open_line() has 604 non-comment lines (error triggered by exceeding 500 lines).  [readability/fn_size] [1]

Reply:
Just ignore it for now. When we merge to master, the linter will update its list of ignored lines.

Reply:
Ok.  But it will trigger for every change of the / around the function then I assume?